### DISCUSSION TITLE
GitHub Pages doesn't use updated CSS file

### DISCUSSION THREAD

**Discussion by Dextrices** [community/community#22048] [Other Feature Feedback, Questions, & Ideas]
<p>I’m using GitHub Pages for my website at this specific site: <a href="https://ludomath.net/LA/vectors.html" rel="nofollow noopener">https://ludomath.net/LA/vectors.html</a>. The problem is that whenever I push new files on GitHub, the CSS files do not update on the site. It just uses an older version. I tried making changes to the HTML file, and that works perfectly, but the CSS file still stays behind and loads an old version. It has been about 1 hour since I pushed the last changes, and it still doesn’t load the new CSS file. </p>
<p>Just to be clear: the CSS file is updated on GitHub, but the actual website does not use the updated file. The CSS file can be found here: <a href="https://github.com/dexterAndre/LudoMath/tree/gh-pages/Common" rel="nofollow noopener">https://github.com/dexterAndre/LudoMath/tree/gh-pages/Common </a></p>
<p>This problem seems to have many causes and fixes, but none have worked for me, so that’s why I’m asking here. I’m pretty new to web development, so please excuse me if I’m missing something obvious. </p>

**Comment by Dextrices** [comment]
<p>So I’ve waited for about 3 hours, and finally the changed were made. I worked with some other files in the meantime, but I don’t know if that was the solution. The problem that the HTML file does update quickly, while the CSS file doesn’t update before 3 hours still persists, and I’m still interested in hearing how I can improve the CSS update speed for GitHub pages. </p>

**Comment by oycyc** [comment]
<p>Hi,</p>
<p>When I first started using GitHub pages, I also had the same experience. It is not GitHub’s fault, instead it is your browser. The solution to this is by going to your browser settings and reset cache and cookies. This should work.</p>
<p>Another way of doing this is by using Git. You make your changes on your local desktop, and see what the website looks like by opening index.html . Then you can git push it to github and be rest assured that it will be like what you saw in your local desktop.</p>

**Comment by Dextrices** [comment]
<p>I have tried deleting the cookies and resetting the cache in Chrome, but it doesn’t help. Also, I’m using both GitHub for Desktop and also directly modifying the file on the GitHub website itself, but none of these work, unfortunately. </p>

**Comment by kimardenmiller** [comment]
<p>About 4 hours and no updates happening. The files updated immediately in the repo, but not on the hosted page.</p>
<p>Finally the page updated…</p>

**Comment by elexie** [comment]
<p>I am experiencing the same problem.</p>
<p>When I use git to upload, the changes are only noticeable in incognito mode. But when I upload directly to github, the changes are noticeable after about 5 minutes.</p>

**Comment by brahimdev** [comment]
<p>same problem too yesterdy it takes 12 hour to updat the steelshet </p>

**Comment by yuyoux** [comment]
<p>Just clear the browsing data in the browser settings and it will work.</p>

**Comment by deeplearning-cmu-10707** [comment]
<p>Same problem for me too.</p>
<p>GitHub pages are not reflecting changes on the github repository. </p>

**Comment by nadiajoyce** [comment]
<p>Hi everyone!</p>
<p>If clearing your cache/removing cookies doesn’t work as explained by <a class="mention" href="/yangciou">@yangciou</a>, I would recommend reaching out to private support with the name and link of the repository you are working with. They should be able to take a closer look.</p>
<p>Cheers!</p>

**Comment by jaikangam** [comment]
<p>It happened to me too. What i understand is you need to clear the cache of your browser. I try on other browser it is working fine. Please clear cache and try hope it works…</p>

**Comment by csteitz01** [comment]
<p>Was having the same issue and Ctrl + f5 did the trick for me (refreshes your cache)</p>

**Comment by dirkroorda** [comment]
<p>No, that is not the solution.</p>
<p>The problem is that the content on gitHub from which the site is built, is not reflected in what <a href="http://github.io" rel="nofollow noopener">github.io</a> serves.</p>
<p>There is an unpredicatble delay which can be seconds or hours.</p>

**Comment by SimonBarendse** [comment]
<p>As the CSS files could also be cached locally, it would be nice if support for <a href="https://github.com/tkareine/jekyll-minibundle" rel="nofollow noopener">https://github.com/tkareine/jekyll-minibundle</a> is added to GitHub pages. minibundle adds a hash of the contents to the filename, so that a different filename is used when changes are made, avoiding the cache altogether for updated files.</p>

**Comment by hesamalavi** [comment]
<p>If you have / before you css file remove it.</p>
<p>Example </p>
<p>&lt;link rel=“stylesheet” href="/main.css" /&gt;</p>
<p>won’t work but</p>
<p>&lt;link rel=“stylesheet” href=“main.css” /&gt;</p>
<p>will work</p>

**Comment by jcubic** [comment]
<p>I have the same problem. This is some kind of url caching on the server, if you change the url by adding <code>?random=1</code> the file is there, but the old urls give old cached result. I’ve just uploaded the file to verifiy in google webmaster tool and I can’t validate because without <code>?anything</code> I have 404.</p>
<p>There should be some kind of <code>flush</code> function in repo settings, where user is able to force update the content. Or maybe using the API.</p>

**Comment by tmetz** [comment, marked as answer]
<p>Hello <a class="mention" href="/dexterandre">@dexterandre</a>!  It definitely sounds like your old version is being cached, and <a class="mention" href="/yangciou">@yangciou</a>, <a class="mention" href="/yuyoux">@yuyoux</a>, <a class="mention" href="/nadiajoyce">@nadiajoyce</a>, <a class="mention" href="/jaikangam93">@jaikangam93</a>, and <a class="mention" href="/csteitz01">@csteitz01</a> are correct in their suggestions.  Sometimes a hard refresh of your browser (as opposed to a standard refresh) is necessary to really clear out the old file.  Below are some instructions for how to do this in various browsers.</p>
<p><strong>Chrome:</strong></p>
<p>Option 1: Hold down Ctrl and click the Reload button (Windows) or Shift and click the Reload button (Mac)<br>
Option 2: Hold down Ctrl and press F5 (Windows) or Cmd and Shift and R (Mac)</p>
<p><strong>Firefox:</strong></p>
<p>Option 1: Hold Ctrl and press the F5 key (Windows) or Shift and click the Reload button (Mac)<br>
Option 2: Hold down Ctrl and Shift and R (Windows) or Cmd and Shift and R (Mac)</p>
<p><strong>Internet Explorer:</strong></p>
<p>Option 1: Hold Ctrl and press the F5 key<br>
Option 2: Hold the Ctrl key and click the Refresh button.</p>
<p><strong>Safari:</strong></p>
<p>Press Shift and click the Reload button</p>
<h2>Additional Resources:</h2>
<p>Here are some resources on the subject of clearing browser cache:</p>
<p>Chrome:<br>
<a href="https://support.google.com/accounts/answer/32050" class="onebox" target="_blank">https://support.google.com/accounts/answer/32050</a></p>
<p>Firefox:<br>
<a href="https://support.mozilla.org/en-US/kb/how-clear-firefox-cache" class="onebox" target="_blank">https://support.mozilla.org/en-US/kb/how-clear-firefox-cache</a></p>
<p>Microsoft Edge:<br>
<a href="https://support.microsoft.com/en-us/help/10607/microsoft-edge-view-delete-browser-history" class="onebox" target="_blank">https://support.microsoft.com/en-us/help/10607/microsoft-edge-view-delete-browser-history</a></p>
<p>Safari:<br>
<a href="https://support.apple.com/guide/safari/clear-your-browsing-history-sfri47acf5d6/mac">https://support.apple.com/guide/safari/clear-your-browsing-history-sfri47acf5d6/mac</a></p>

**Reply by DmitriyPast** [reply]
Thanks it works
